### PR TITLE
Fix some install paths, and provide help menu entry

### PR DESCRIPTION
**Pull request by mwilck**
These fixes were created while I was working on the openSUSE package lately. Well, I guess the second one is more a feature addition than a bug fix, although I like it a lot. I can create another PR without it if you prefer.

### CODE REVIEW

**Comment by ederag** [comment]
Sounds interesting, thanks !
I would like to merge the help stuff as a whole so yes,
please move out the bash-completion commit to another PR, since it is unrelated.
There will be some delay, as I'm going to set up a virtualbox
to ensure that installation do not break on ubuntu.

**Comment by ederag** [comment]
In `lubuntu-18.10` for the development version, 
when the help files are not installed 
```python
Traceback (most recent call last):
  File "/media/sf_python/hamster/src/hamster/overview.py", line 546, in on_help_clicked
    gdk.CURRENT_TIME)
gi.repository.GLib.Error: g-io-error-quark: The specified location is not supported (15)
```
Having an error is normal (the help files are missing), 
but the error message could be clearer.

By the way, the help files are not installed by 
```bash
./waf configure build --prefix=/usr && sudo ./waf install
```
even when all dependencies are satisfied :
`python3-gi-cairo gir1.2-gconf-2.0 python3-xdg python3-dbus gettext intltool gnome-doc-utils docbook-utils`
How should the help files be installed ?

**Comment by mwilck** [comment]
> In `lubuntu-18.10` for the development version,
> when the help files are not installed
> 
> ```python
> Traceback (most recent call last):
>   File "/media/sf_python/hamster/src/hamster/overview.py", line 546, in on_help_clicked
>     gdk.CURRENT_TIME)
> gi.repository.GLib.Error: g-io-error-quark: The specified location is not supported (15)

Do you have the GNOME help browser installed (yelp)? It's responsible to display help files.

But yeah, I guess we need to catch that error, similar to how it's done for in the "export" code when opening non-HTML files. Althougth the generic "except: pass" there looks a bit too catch-all to my taste.

> By the way, the help files are not installed by
> 
> ```shell
> ./waf configure build --prefix=/usr && sudo ./waf install
> How should the help files be installed ?

I was looking at the "waf configure" output which complained about some tools being missing. See [the build log](https://build.opensuse.org/build/devel:openSUSE:Factory:Apps/openSUSE_Tumbleweed/x86_64/hamster-time-tracker/_log). I added xml2pod (and xsltproc, but I don't think it's necessary), which caused the help files to be generated and installed. Waf also looks for docbook2man, but AFAICS it doesn't do anything with it, and docbook2man (at least on SUSE) pulls in an enormous list of build dependencies via tetex, so I did not add it.

**Comment by mwilck** [comment]
> Do you have the GNOME help browser installed (yelp)? It's responsible to display help files.

Well it's "yelp" under GNOME. Other desktops may have their own help display tools. But the install location and the file format are cross-desktop AFAICT.

**Comment by mwilck** [comment]
I pushed another patch to handle the "help browser not available" problem, and dropped the bash-completion thing for now.

File: src/hamster/overview.py [hunk 1]
```diff
@@ -538,6 +542,21 @@ def on_timeout(self):
         # The timeout will stop if returning False
         return True
 
+    def on_help_clicked(self, menu):
+        uri = "help:hamster-time-tracker"
+        try:
+            gtk.show_uri(gdk.Screen().get_default(), uri,
```
Comment (ederag) [inline src/hamster/overview.py]:
```suggestion
            gtk.show_uri(None, uri,
```
`None` seems the standard way to say "default location"
https://lazka.github.io/pgi-docs/#Gtk-3.0/functions.html#Gtk.show_uri
https://programtalk.com/python-examples/gi.repository.Gtk.show_uri/
It does not seem possible to make a suggestion on several lines,
but of course it should be `gtk.show_uri(None, uri, gdk.CURRENT_TIME)`

File: src/hamster/overview.py [hunk 2]
```diff
@@ -538,6 +542,21 @@ def on_timeout(self):
...
+                         gdk.CURRENT_TIME)
+        except gi.repository.GLib.Error:
+            msg = sys.exc_info()[1].args[0]
+            dialog = gtk.MessageDialog(self.window, 0, gtk.MessageType.ERROR,
+                                       gtk.ButtonsType.CLOSE,
+                                       _("Failed to open {}").format(uri))
+            fmt = _('Error: "{}" - is a help browser installed on this Computer?')
```
Comment (ederag) [inline src/hamster/overview.py]:
```suggestion
            fmt = _('Error: "{}" - is a help browser such as yelp installed on this Computer?')
```
Mentioning `yelp` to further guide the user ?

File: src/hamster/overview.py [hunk 3]
```diff
@@ -557,7 +576,8 @@ def on_report_chosen(widget, format, path):
                 webbrowser.open_new("file://%s" % path)
             else:
                 try:
-                    gtk.show_uri(gdk.Screen(), "file://%s" % os.path.split(path)[0], 0)
+                    gtk.show_uri(gdk.Screen().get_default(),
```
Comment (ederag) [inline src/hamster/overview.py]:
```suggestion
                    gtk.show_uri(None,
```
As above,
`None` seems the standard way to say "default location"
https://lazka.github.io/pgi-docs/#Gtk-3.0/functions.html#Gtk.show_uri
https://programtalk.com/python-examples/gi.repository.Gtk.show_uri/
It does not seem possible to make a suggestion on several lines,
but of course it should be `gtk.show_uri(None, "file://%s" % path, gdk.CURRENT_TIME)`

**Comment by ederag** [comment]
> How should the help files be installed ?

Good news: there was only a cache issue. The safest build is
```bash
./waf clean && ./waf distclean && ./waf configure build --prefix=/usr && sudo ./waf install
```
(both `clean` and `distclean` were necessary to get the help built)

File: src/hamster/overview.py (hunk 2 above)
Comment (mwilck) [reply src/hamster/overview.py]: I don't think `yelp` is the only possibility. The KDE Help browser is another I guess. I wouldn't want to be too specific, or biased towards a certain environment. Next year "yelp" may have been replaced by something else in GNOME.

File: src/hamster/overview.py (hunk 1 above)
Comment (mwilck) [reply src/hamster/overview.py]: Yes, "None" works, too. It comes down to exactly the same thing AFAICT.

File: src/hamster/overview.py (hunk 2 above)
Comment (mwilck) [reply src/hamster/overview.py]: Moreover, the name "yelp" is also pretty much hidden in GNOME environments. I believe that using such specific names is pretty much deprecated in GUI environments. "Help browser" should mean something to most users, "yelp" definitely doesn't. It's true that users need to figure out which package they need to install to get the help browser - but honestly that's GNOMEs or distributor's job, not ours.

File: src/hamster/overview.py (hunk 1 above)
Comment (mwilck) [reply src/hamster/overview.py]: If you read that documentation, you'd expect Gdk.Screen() to be correct as first arg to Gtk.show_uri(). But it crashes reliably, I tested that in various enviroments. I wonder if we have spotted a problem in PyGTK (or GI, for that matter). But I don't care that much, actually.

Comment (mwilck) [reply src/hamster/overview.py]: The types of gdk.Screen() and gdk.Screen.get_default() are different:

    >>> gdk.Screen()
    <Gdk.Screen object at 0x7f7e89bd4f00 (GdkScreen at 0x56502c04eb60)>
    >>> gdk.Screen().get_default()
    <GdkX11.X11Screen object at 0x7f7e89b8d690 (GdkX11Screen at 0x56502bf36000)>

**Comment by mwilck** [comment]
I made the "None" change now. Feel free to modify the error message as you see fit.

File: src/hamster/overview.py (hunk 2 above)
Comment (ederag) [reply src/hamster/overview.py]: That makes sense. We'll just add a line in the [dependencies section](https://github.com/projecthamster/hamster#dependencies).

Comment (mwilck) [reply src/hamster/overview.py]: Make sure to make it a "weak" dependency though. In distro speak, it would be a "Recommends:", not a "Requires:".

Comment (ederag) [reply src/hamster/overview.py]: The KDE help browser `khelpcenter` does not seem to read these `mallard` pages.
(they do not show up when searching for "hamster").
Yelp seems to be the only solution currently. No big deal; it is both nice and lightweight.
https://github.com/mwilck/hamster/pull/3 starts with a proposition about that.

File: src/hamster/overview.py (hunk 1 above)
Comment (ederag) [reply src/hamster/overview.py]: Interesting, but the `None` version you pushed seems fine, thanks.

**Comment by ederag** [comment]
Comments left above; let's continue in https://github.com/mwilck/hamster/pull/3 for now.

**Comment by mwilck** [comment]
Merged your suggestion, your turn now ;-)

**Comment by ederag** [comment]
Well thanks, but I was not done :smile: 
The help pages are very good, but still there is plenty of work to update them first
(broken links, some out-of-date stuff, ...). 
It will take a while (hence the work in progress header in my PR).

**Comment by mwilck** [comment]
> The help pages are very good, but still there is plenty of work to update them first

Alright ... I thought that could be done in a later step. But fine with me this way.

**Comment by ederag** [comment]
Thanks for your comprehension.
The help pages are very important, and this will make a nice homogeneous bunch of commits.

**Comment by ederag** [comment]
@mwilck Thanks for bringing the help back to life !

**Pull request merged**